internal/core/src/mmap/Column.h

```diff
@@ -918,7 +918,15 @@ class SingleChunkArrayColumn : public SingleChunkColumnBase {
     void
     Append(const Array& array, bool valid_data = false) {
         indices_.emplace_back(data_size_);
-        element_indices_.emplace_back(array.get_offsets());
+        lens_.emplace_back(array.length());
+        if (IsVariableDataType(array.get_element_type())) {
+            element_indices_.emplace_back(
+                array.get_offsets_data(),
+                array.get_offsets_data() + array.length());
+        } else {
+            element_indices_.emplace_back();
+        }
+
```

Comment:
have to emplace empty vector for inner indices to keep the length align with indices